### DISCUSSION TITLE
Does Github Pull request and Code Insight Plugin support multiple instance of github ?

### DISCUSSION THREAD

**Discussion by mdoshi111** [RoadieHQ/roadie-backstage-plugins#775] [General]
For our usecase in our company we have 2 different versions of Github.

app-config.yaml file

integrations:
  github:
    - host: github.com
      apiBaseUrl: ${github_apiBaseUrl}
      token: ${github_token}
    - host: github.companyname.com
      apiBaseUrl: ${github_enterprise_apiBaseUrl}
      token: ${github_enterprise_token}

While using github pull request and code insight plugin , It only loads data from github.com domain and not loading repo data for github.companyname.com domain.

Is there a way we can load repo data from both of the github URL?

**Comment by ltdonner-bah** [comment]
We're also interested in this - our use case is that we have a GitHub enterprise integration and right now we also have public github.com components registered as well. For those components, it tries to send the GH enterprise token regardless of if it's a GHE location or a public GitHub one.

It would be nice if we could have the plugins log the user in to get tokens from the appropriate GitHub provider and send that token with the request. 

Right now for the public github.com repos, we get a 401 back with a "Bad credentials" error.
![image](https://github.com/RoadieHQ/roadie-backstage-plugins/assets/141174159/a7497e78-e5c3-4070-a1b8-e68f51c996a3)

There's a middle step for public github.com repos where we can just check if the location for the component shares a host with an existing integration, but that runs up against rate limiting pretty quick since it's an unauthenticated request. Ideally we'd get the user's public github token and send that along instead of trying to use the GHE token for everything
